File: src/test/java/br/com/blz/testjava/dto/WarehouseDtoTest.java
package br.com.blz.testjava.dto;

import br.com.blz.testjava.domain.Warehouse;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class WarehouseDtoTest {

    @Test
    public void createWarehouseDtoTest() {
        Warehouse warehouse = new Warehouse();
        warehouse.setLocality("SP");
        warehouse.setQuantity(1);
        warehouse.setType("PHYSICAL_STORE");

        WarehouseDto warehouseDto = new WarehouseDto(warehouse);

        String localityExpected = "SP";
        int quantityExpexted = 1;
        String typeExpected = "PHYSICAL_STORE";

        assertEquals("Localidade não compatível", localityExpected, warehouseDto.getLocality());
        assertEquals("Tipo não compatível", typeExpected, warehouseDto.getType());
        assertEquals("Quantidade não compatível", quantityExpexted, warehouseDto.getQuantity().intValue());
    }

}

File: src/test/kotlin/br/com/blz/testjava/controller/dto/WarehouseDTOTest.kt
package br.com.blz.testjava.controller.dto

import br.com.blz.testjava.domain.entities.Warehouse
import br.com.blz.testjava.domain.entities.WarehouseType
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

internal class WarehouseDTOTest {

  @Test
  fun `Must convert a Warehouse entity to WarehouseDTO`() {
    val warehouse = Warehouse("SP", 10, WarehouseType.ECOMMERCE)
    val warehouseDTO = WarehouseDTO(warehouse)

    assertEquals(warehouseDTO.locality, warehouse.locality)
    assertEquals(warehouseDTO.quantity, warehouse.quantity)
    assertEquals(warehouseDTO.type, warehouse.type)
  }

}
